script.py

```
@@ -276,6 +291,39 @@ def run_async(coro: Coroutine) -> asyncio.Future | None:
+    </div>
+    <div id="dss-phases"></div>
+</div>"""
+
+
+def custom_js():
+    """Returns JavaScript for SSE client - injected via TGWUI's custom_js hook."""
+    import os
+    from jinja2 import Environment, FileSystemLoader
+    js_path = os.path.join(os.path.dirname(__file__), "ui")
+    try:
+        env = Environment(loader=FileSystemLoader(js_path))
```

Ruff (0.15.9)

[error] By default, jinja2 sets `autoescape` to `False`. Consider using `autoescape=True` or the `select_autoescape` function to mitigate XSS vulnerabilities.

(S701)